docs/01-programs/02-candy-machine/09-available-guards/05-freeze-sol-payment.md

```diff
@@ -0,0 +1,223 @@
+
+</div>
+</AccordionItem>
+</Accordion>
+
+## Mint Settings
+
+_The Freeze Sol Payment guard does not need Mint Settings._
+
+However, if you’re planning on constructing instructions without the help of our SDKs, you will need to add the configured destination address to the remaining accounts of the mint instruction. See the [Candy Guard’s program documentation](https://github.com/metaplex-foundation/mpl-candy-guard#freezesolpayment) for more details.
+
+## Route Instruction
+
+The Freeze Sol Payment route instruction supports the following features.
+- [Initialize the Freeze Escrow](#initialize-the-freeze-escrow)
+- [Thaw a Frozen NFT](#thaw-a-frozen-nft)
+- [Unlock Funds](#unlock-funds)
+
+### Initialize the Freeze Escrow
+
+*Path: `initialize`*
+
+When using the Freeze Sol Payment guard, we must initialize the Freeze Escrow account before minting can start. This will create a PDA account derived from the Destination attribute of the guard's settings.
+
+The Freeze Escrow PDA account will keep track of several parameters such as:
+- How many Frozen NFTs were minted through this guard.
+- When was the first Frozen NFT minted via this guard as the Freeze Period starts counting after that.
+
+When initializing this Freeze Escrow account, we must provide the following arguments to the route instruction of the guard:
+- **Path** = `initialize`: Selects the path to execute in the route instruction.
+- **Period**: The amount of time in seconds that the Freeze Period should last. This can be a maximum of 30 days (2,592,000 seconds) and it will start from the very first Frozen NFT minted via this guard. The Freeze Period provides a safety mechanism to ensure Frozen NFTs can eventually be thawed even if the Candy Machine never mints out.
+- **Candy Guard Authority**: The authority of the Candy Guard account as a Signer.
+
+![CandyMachinesV3-GuardsFreezeSolPayment1.png](/assets/candy-machine-v3/CandyMachinesV3-GuardsFreezeSolPayment1.png#radius)
+
+Last but not least, the Freeze Escrow PDA account will receive the funds of all Frozen NFTs minted through this guard.
+
+![CandyMachinesV3-GuardsFreezeSolPayment2.png](/assets/candy-machine-v3/CandyMachinesV3-GuardsFreezeSolPayment2.png#radius)
+
+<Accordion>
+<AccordionItem title="JS SDK" open={true}>
+<div className="accordion-item-padding">
+
+In the example below, we initialize the Freeze Escrow account with a maximum Freeze Period of 15 days and we use the current identity as the Candy Guard authority.
+
+```tsx
+await metaplex.candyMachines().callGuardRoute({
+  candyMachine,
+  guard: 'freezeSolPayment',
+  settings: {
+    path: 'initialize',
+    period: 15 * 24 * 60 * 60, // 15 days.
```

Comment:
Would it make sense to add an utility function to convert days/hours to seconds? E.g., `secondsFromDays`, `secondsFromHours`.

Reply:
I wanted to do that on the SDK initially but thought it's maybe too generic and people would already have their way of translating time within their framework. I'm happy to revisit this later though.